### PR TITLE
add Integration Test for Gradle Plugin

### PR DESCRIPTION
**Pull request by vorburger**
#1396

### CODE REVIEW

File: devtools/gradle-it/build.gradle [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    }
+    dependencies {
+        classpath fileTree(dir: '../gradle/build/libs', include: 'quarkus-gradle-plugin-999-SNAPSHOT.jar')
+        classpath fileTree(dir: '../gradle/target/dependencies/compile', include: '*.jar')
+    }
+}
+
+repositories {
+     mavenCentral()
+     maven { url 'http://repository.jboss.org/nexus/content/groups/public' }
```
Comment (stalep) [inline devtools/gradle-it/build.gradle]: not sure if this line is needed?

File: devtools/gradle-it/pom.xml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                <artifactId>exec-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>gradle-integration-test</id>
+                        <phase>prepare-package</phase>
+                        <configuration>
+                            <executable>../gradle/gradlew</executable>
+                            <arguments>
+                                <argument>build</argument>
+                                <argument>quarkus-build</argument>
```
Comment (stalep) [inline devtools/gradle-it/pom.xml]: quarkus-build requires build, so running build first is not needed

**Comment by stuartwdouglas** [comment]
This failure looks related.

File: devtools/gradle/settings.gradle [hunk 3]
```diff
@@ -1 +1 @@
-rootProject.name = projectName
+rootProject.name = 'quarkus-gradle-plugin'
```
Comment (stalep) [inline devtools/gradle/settings.gradle]: projectName is set in gradle.properties, if we change this we should remove that file i guess.

**Comment by stalep** [comment]
i did not get any failures when checking out the branch and doing a mvn clean install -Dno-native

**Comment by vorburger** [comment]
> This failure looks related.

[BIWL](https://memegenerator.net/instance/65488750/hindenburg-but-it-worked-locally) ... must be because:

```
[INFO] --- exec-maven-plugin:1.6.0:exec (gradle) @ quarkus-gradle-plugin ---
(...)
[INFO] skipping execute as per configuration
```

This happens if it runs with `-Dskip.gradle.build=true` ... it took me a moment to figure out how come that's set in the project's CI ... it's because of this beauty at the end of `devtools/gradle/pom.xml` : 

```
        <profile>
            <id>java8</id>
            <activation>
                <jdk>[8,)</jdk>
            </activation>
            <properties>
                <skip.gradle.build>true</skip.gradle.build>
            </properties>
        </profile>
```

@stalep what's the reason for this? `quarkus-gradle-plugin`  (locally tested) seems to (now?) build just fine even under Java 11... My preference would therefore be to remove this, therefore so suggested in #1425. If that is not acceptable, then we just add the same in `gradle-it/pom.xml` instead, that will do the trick. Thoughts?

**Comment by stalep** [comment]
> @stalep what's the reason for this? `quarkus-gradle-plugin` (locally tested) seems to (now?) build just fine even under Java 11... My preference would therefore be to remove this, therefore so suggested in #1425. If that is not acceptable, then we just add the same in `gradle-it/pom.xml` instead, that will do the trick. Thoughts?
hm, that do not need to be there. it was just brought forward from the example i used to trigger gradle from maven.

File: devtools/gradle-it/build.gradle (hunk 1 above)
Comment (vorburger) [reply devtools/gradle-it/build.gradle]: You're right it's not, so I've removed it.

File: devtools/gradle/settings.gradle (hunk 3 above)
Comment (vorburger) [reply devtools/gradle/settings.gradle]: I'm a little bit fuzzy on which of these Gradle files is for what exactly... :smile: and I had to change this for the built JAR from the plugin to have a better name than projectName.jar, so I would rather not touch this anymore, if that is acceptable to get at least a first version of this IT in?

File: devtools/gradle-it/pom.xml (hunk 2 above)
Comment (vorburger) [reply devtools/gradle-it/pom.xml]: Yup; I've change it. And that reminds me of something we discussed face to face that day => see #1442.

**Comment by stuartwdouglas** [comment]
@stalep I think this looks ok to me, if @stalep approves we can merge it

**Review by stalep — APPROVED** [review]

**Pull request merged**